packages/core/entity/FileExplorerURL/index.ts

```diff
@@ -48,6 +48,17 @@ export const DEFAULT_AICS_FMS_QUERY: FileExplorerURLComponents = {
     sortColumn: new FileSort(AnnotationName.UPLOADED, SortOrder.DESC),
 };
 
+export const getNameAndTypeFromSourceUrl = (dataSourceURL: string) => {
+    const uriResource = dataSourceURL.substring(dataSourceURL.lastIndexOf("/") + 1).split("?")[0];
+    const name = `${uriResource} (${new Date().toLocaleDateString()} ${new Date().toLocaleTimeString()})`;
+    let extensionGuess = uriResource.split(".").pop();
+    if (!(extensionGuess === "csv" || extensionGuess === "json" || extensionGuess === "parquet")) {
+        console.warn("Guess that the source is a CSV file since no extension easily found");
```

Comment:
nit: Maybe "Assuming the source is csv since no extension was recognized"

Reply:
Changed in [f5bcbcf](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/117/commits/f5bcbcf033ba64d00b3bbddf185c30e73f30ce1b)[f5bcbcf](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/117/commits/f5bcbcf033ba64d00b3bbddf185c30e73f30ce1b)